### PR TITLE
Haohui - Gives the user permission to add new user permissions role

### PR DESCRIPTION
**Pull request by Lin1404**
# Description
image

## Related PRS (if any):
To test this backend PR you need to checkout the #1307 frontend PR.
…

## How to test:
1. check into current branch
2. do npm run build and npm run start to run this PR locally
3. log as owner user call user A
4. go to dashboard→ Other Links→ Permissions Management→ Manage User Permissions
5. input another user name, call user B
6. find the “Add New User Permissions Role”, click the add button
7. scroll down and click submit button
8. log out and login as user B(if already logged in, make sure to log out and log back in)
9. go to dashboard→ Other Links→ Permissions Management→ Add New Role
10. create new role, input the role name, select permissions, then submit
11. the new role show in the "user roles" table

## Note:
Include the information the reviewers need to know.

### CODE REVIEW

**Review by wantingxu7 — APPROVED** [review]
Hi Haohui, good work, the function works well. Left detailed comment in FE

**Review by JYXiao-2021 — APPROVED** [review]
Hi, I left my comment at FE.

**Review by navneeeth — APPROVED** [review]
Hello! The PR works as intended. Good job! Commented a detailed review on the frontend PR.

**Review by vikrambadhan — APPROVED** [review]
Hi, the PR works well. Left my comment on FE

**Review by Shiwani99 — APPROVED** [review]
Hi,
Great work. Added comments at frontend PR#1307.

**Review by Bhagyashree444 — APPROVED** [review]
Well done! Works as expected. I've dropped a comment on the frontend PR#1307.

**Review by StrawberryCalpico — CHANGES_REQUESTED** [review]
Hi! I just reviewed your PR, it seems like there's a conflict with the current branch, so that I can't find the add new user permission role. the video was in  [#1307](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1307#pullrequestreview-1665122487)

**Review by malikjahanzaib — CHANGES_REQUESTED** [review]
Hi @Lin1404, it seems this branch has some conflicts. I have checked out both FE and BE branches. When I try to go to "dashboard→ Other Links→ Permissions Management→ Manage User Permissions" from an owner account, I search for the user to assign the permission but there appears no “Add New User Permissions Role”. Below is the test recording.

Left detailed review on FE.

**Comment by Cloid** [comment]
Left comments in FE #1307.

**Comment by ChirayuRai** [comment]
Hey! I added my comments in FE #1307

File: src/controllers/rolesController.js [hunk 1]
```diff
@@ -10,8 +10,11 @@ const rolesController = function (Role) {
   };
 
   const createNewRole = async function (req, res) {
-    if (!await hasPermission(req.body.requestor.role, 'postRole')) {
-      res.status(403).send('You are not authorized to create new roles.');
+    if (
+      !(await hasPermission(req.body.requestor.role, "postRole")) &&
+      !req.body.requestor.permissions?.frontPermissions.includes("postRole")
+    ) {
+      res.status(403).send("You are not authorized to create new roles.");
```
Comment (nathanah) [inline src/controllers/rolesController.js:17]: This change isn't needed as of PR547.
https://github.com/OneCommunityGlobal/HGNRest/pull/547

**Comment by palakgosalia** [comment]
Hey, I left comments in PR 1307. Thank you

**Review by Alforoan — COMMENTED** [review]
I left my comment on the fe pr #1307

**Pull request closed without merging**